extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/health/DataSourceHealthCheck.java

```diff
@@ -40,10 +41,12 @@ protected void init() {
         }
         DataSourceSupport support = Arc.container().instance(DataSourceSupport.class)
                 .get();
+        AgroalDataSourceSupport agroalSupport = Arc.container().instance(AgroalDataSourceSupport.class)
+                .get();
         Set<String> names = support.getConfiguredNames();
```

Comment:
I wonder if you shouldn't just rely exclusively on `agroalSupport.entries` and get rid of `support.getConfiguredNames()` altogether. `support.getConfiguredNames()` is sometimes returning non-agroal datasource names (which you're working around here), and I've noticed before (#37779) that it may not even return the name of all configured Agroal datasources:

https://github.com/quarkusio/quarkus/blob/5bbae6fc32ca65188b91d0f86db987fde727332f/extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourceRecorder.java#L21-L29

Reply:
Yeah, you're right. I dropped it in a second commit so that we don't introduce any more confusion but I will only backport the first one.